src/OrleansRuntime/GrainDirectory/ClientObserverRegistrar.cs

```diff
@@ -102,6 +103,7 @@ internal void ClientDropped(GrainId clientId)
 
         private async Task OnClientRefreshTimer(object data)
         {
+            if (gateway == null) return;
```

Comment:
this should not be possible since you are only starting the timer after you assign the gateway, or am I missing something?